src/StreamRequest.jl

```diff
@@ -68,8 +76,8 @@ function request(::Type{StreamLayer}, io::IO, request::Request, body;
         end
 
     catch e
-        if e isa CompositeException
-           e = first(e.exceptions).ex
+        if write_error != nothing
+            throw(write_error)
```

Comment:
Is this changing behavior from before? Didn't we pass on to the next `if` block before and check is it's a `isioerror`. Why change that now?

Reply:
hmm, looking at this again, I think there is more to it...

Assume that `writebody` is sending some huge message body, and the server sends an error with `Connection: "close` (https://tools.ietf.org/html/rfc7230#section-6.5).
`readbody` will return, and `isaborted()` will be true on line 72.
So, close gets called and the `writebody` task stop with an IO error, which is stored in `write_error`.
However, the `catch` block at 79 will never be reached, because there is no exception to catch  in the main task, so `write_error` will not be used at all. This makes sense in this case because the main task has decided to kill the `writebody` task and doesn't care about any subsequent exceptions that it might throw.

I think we can reduce the `catch` block to just:
```julia
    catch e
        if write_error != nothing
            throw(write_error)
        else
            rethrow(e)
        end
    end
```

This makes the intent of the code a bit easier to follow, i.e: If there is an error while reading the response, check if the root-cause was an error sending the request, otherwise just rethrow the reading error.